src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,146 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    public static class Node<T> {
+        private T value;
+        private Node<T> prev;
+        private Node<T> next;
+
+        public Node(T value) {
```

Comment:
[When creating a new Node, pass all the fields to Node constructor.](https://mate-academy.github.io/jv-program-common-mistakes/java-core/collections/linked-list.html#when-creating-a-new-node-pass-all-the-fields-to-node-constructor)